odoo/models.py

```diff
@@ -6160,22 +6160,35 @@ def _get_placeholder_filename(self, field=None):
         return 'placeholder.png'
 
     def _populate_factories(self):
-        """ Return a list of pairs ``(field_name, factory)``, ``factory`` is a function that creates
-        a generator of values (dict of field values).  Its API suggested by::
+        """ Return a list of pairs ``(field_name, factory)``.
 
-            # generator yield values by iterating on base_generator and adding a value
-            # for the given field in the dicts yielded by base_generator
-            generator = factory(base_generator, field_name, model_name)
-            # suggested usage
-            for values in generator:
-                model.create(values)
+        ``factory`` is a generator of values (dict of field values).
```

Comment:
Seems like this would benefit from more explanations (also it's not necessarily a generator, and it's a wrapper for an other iterator), what's the "dict of field values", is it only supposed to fill the field name it's initialised with or can it do whatever? What's the mutation contract with respect to the values it retrieves from the iterator it wraps?

Reply:
I didn't see any strict rule enforcing only one value should be updated, even if it's not really the expected behavior.